docs/developer-guides/migration-guide-1.4.0.md

```diff
@@ -0,0 +1,114 @@
+5. Upgrade the runtime of the chain using the newly built WASM (it can be found in `$CARGO_TARGET_DIR/release/wbuild/<runtime name>/<runtime name>.compressed.wasm`) and the
+extrinsic `system/setCode`.
+
+After these steps the chain should be running the runtime using Partner Chains SDK v1.3.1. This version is
+fully backwards-compatible and is a preparatory step towards v1.4.0
+
+### Release new version using PC-SDK v1.4.0
+
+This version of the node is backwards-compatible with the runtime in versions 1.3.x. and is needed to support
+the runtime in version v1.4.0.
+
+1. Update all Partner Chains SDK dependencies in your `Cargo.toml` file to `v1.4.0`
+2. Modify your runtime and node code to be compatible with PC-SDK version v1.4.0. Consult the release notes for v1.4.0
+for required changes.
+3. Release a new version of your node. This step depends on your release process.
+4. Upgrade nodes running the chain to the newly released version. This step depends on your deployment process.
+
+After this step, the network should be composed of nodes v1.4.0 but the runtime would remain at v1.3.1, and be ready for the matching
+runtime upgrade to v1.4.0.
+
+### New main chain state setup
+
+Because of the changes to the smart contracts and their on-chain data, a completely new set
+of main chain data needs to be set up on Cardano.
+
+1. Pick the initial governance authority.
+This can be a newly generated or already existing main chain key set controlled by the Partner Chain authority.
+2. Run the `prepare-configuration` command of `partner-chains-cli` (v1.4.0) in a fresh directory.
+This step will involve selecting the _genesis UTXO_ to be spent intializing the governance mechanism.
+Save the `partner-chains-cli-chain-config.json` file produced by this step.
+3. Run the `setup-main-chain-state` command of `partner-chains-cli`, setting up the D-param and permissioned candidates.
+
+After these steps, the new Partner Chain will be initialized on Cardano.
+
+### Re-register candidates
+
+At this point all SPOs willing to continue participating in the chain must re-register themselves with the newly created Partner Chain on Cardano.
+
+1. Distribute the `partner-chains-cli-chain-config.json` file produced in the previous section to the SPOs.
+2. Each individual SPO must use it to go through the registration process
+by running the `register1`, `register2`, `register3` commands of `partner-chains-cli` (v1.4.0).
+
+After this step the SPO should be ready to be included in post-migration committee selection.
+
+### Upgrade the runtime to v1.4.0
+
+This step will switch the Partner Chain from observing the old on-chain Cardano data to the new one created in the
+previous steps. These steps should be performed after *at least two* Cardano epochs after SPO re-registrations were performed.
+
+
+1. get new addresses by running:
+```
+./pc-contracts-cli addresses \
+-k payment.skey
+--network testnet \
+--genesis-utxo <genesis utxo> | jq
```

Comment:
How does user know genesis utxo?

It is in partner-chains-cli-chain-config.json now. Or we can also replace this with `jq` over chain-spec.json.

```
jq .genesis.runtimeGenesis.config.sidechain.genesisUtxo chain-spec.json
"0a67554be063e701a205455e0f92d054bcc932f845431af589a00699e1f257a5#1"
➜  spo-v1-3 jq .genesis.runtimeGenesis.config.sessionCommitteeManagement.mainChainScripts chain-spec.json
{
  "committee_candidate_address": "0x616464725f7465737431777266706c7630683264667565776c7768716a6b677178716c676e666d70713463617633653937326b6d68363672716a733333746c",
  "d_parameter_policy_id": "0xad7466d331b9f20148827ff37483affc33be809e32efefd700b5685b",
  "permissioned_candidates_policy_id": "0x769781b1ce42f7a0db4639094d030371ee271415099702a6ecd57f51"
}
```

Reply:
I just added an instruction to get these values from the config